### PR TITLE
Impl/state controller ut for wep apps

### PR DESCRIPTION
**Pull request by AByzhynar**
Fixes FORDTCN-5917

This PR is **[ready]** for review.

### Risk
This PR makes **[no]** API changes.

### Testing Plan
NA

### Summary
Added webengine projection apps as media and non-media to the existing unit test cases

### Tasks Remaining:
- [ ] [Add unit test case for new WEP apps in resuming state]

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/test/application_manager_impl_test.cc [hunk 1]
```diff
@@ -2018,6 +2018,214 @@ TEST_F(
...
+  EXPECT_EQ(mobile_apis::VideoStreamingState::NOT_STREAMABLE,
+            wep_app->video_streaming_state());
+  EXPECT_EQ(mobile_apis::AudioStreamingState::NOT_AUDIBLE,
+            wep_app->audio_streaming_state());
+  EXPECT_EQ(mobile_apis::SystemContext::SYSCTXT_MAIN,
+            wep_app->system_context(main_window_id));
+
+  // Deactivate Webengine projection non-media app
+  // to check its new HMI level and audio & video streaming states
+  app_manager_impl_->state_controller().DeactivateApp(wep_app, main_window_id);
```
Comment (LitvinenkoIra) [inline src/components/application_manager/test/application_manager_impl_test.cc:2096]: @AByzhynar Is it possible to split this test into several separate tests?

Comment (mvorobio) [reply src/components/application_manager/test/application_manager_impl_test.cc:2096]: @AByzhynar the tests could be splitted into 4 parts to check Register, Activate, Deactivate and  Exit separately. Also message construction and state checks could be moved to separate functions.

Comment (AByzhynar) [reply src/components/application_manager/test/application_manager_impl_test.cc:2096]: @mvorobio Activate, Deactivate and Exit separately have no sense without Registration? Make an example how to fo it without registration.

Comment (mvorobio) [reply src/components/application_manager/test/application_manager_impl_test.cc:2096]: @AByzhynar  you're right, Activate, Deactivate and Exit tests should contain registration as well but without additional checks. I think, the tests would be clearer.

File: src/components/application_manager/test/application_manager_impl_test.cc [hunk 2]
```diff
@@ -2018,6 +2018,214 @@ TEST_F(
...
+  rai[strings::msg_params][strings::app_name] = kAppName;
+  rai[strings::msg_params][strings::app_hmi_type] = "WEB_VIEW";
+  rai[strings::msg_params][strings::is_media_application] = false;
+  rai[strings::msg_params][strings::sync_msg_version][strings::minor_version] =
+      APIVersion::kAPIV2;
+  rai[strings::msg_params][strings::sync_msg_version][strings::major_version] =
+      APIVersion::kAPIV6;
+
+  rai[strings::params][strings::protocol_version] =
+      protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_2;
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/application_manager_impl_test.cc:2048]: @AByzhynar I believe it's a good candidate for a separate function in the `ApplicationManagerImplTest` like `GenerateWEPRaiRequest` which returns you a generated smart object. You may pass some specific values as input params

File: src/components/application_manager/test/application_manager_impl_test.cc [hunk 3]
```diff
@@ -2018,6 +2018,214 @@ TEST_F(
...
+  app_manager_impl_->SetPluginManager(rpc_plugin_manager);
+  auto wep_app = app_manager_impl_->RegisterApplication(rai_ptr);
+  wep_app->set_is_media_application(false);
+
+  EXPECT_EQ(protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_2,
+            wep_app->protocol_version());
+  EXPECT_EQ(APIVersion::kAPIV2, wep_app->version().min_supported_api_version);
+  EXPECT_EQ(APIVersion::kAPIV6, wep_app->version().max_supported_api_version);
+
+  const WindowID main_window_id = 0;
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/application_manager_impl_test.cc:2063]:
```suggestion
  const WindowID main_window_id = mobile_apis::PredefinedWindows::DEFAULT_WINDOW;
```

File: src/components/application_manager/test/application_manager_impl_test.cc [hunk 4]
```diff
@@ -2018,6 +2018,214 @@ TEST_F(
...
+  // Initial HMI level, audio, video streaming states and system context
+  // set after app registration
+  EXPECT_EQ(mobile_apis::HMILevel::INVALID_ENUM,
+            wep_app->hmi_level(main_window_id));
+  EXPECT_EQ(mobile_apis::VideoStreamingState::INVALID_ENUM,
+            wep_app->video_streaming_state());
+  EXPECT_EQ(mobile_apis::AudioStreamingState::INVALID_ENUM,
+            wep_app->audio_streaming_state());
+  EXPECT_EQ(mobile_apis::SystemContext::SYSCTXT_MAIN,
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/application_manager_impl_test.cc:2072]: @AByzhynar I am thinking about moving this checks to a separate function as I see it at least 4 times in your test. However I don't really like a function which accepts app ptr + 4 more input params. 
Probably you may create a separate function `MakeHmiState` which accepts level, a/v state, context and returns `HmiHmiStatePtr` with all configured fields. And another function `CompareApplicationHmiState` which accepts app ptr and hmi state ptr and do the comparison like that. Such approach may make your test fixture shorter and simpler

File: src/components/include/application_manager/state_controller.h [hunk 5]
```diff
@@ -223,6 +223,14 @@ class StateController {
    * @param app pointer to application to be exited
    */
   virtual void ExitDefaultWindow(ApplicationSharedPtr app) = 0;
+
+  /**
+   * @brief Sets BACKGROUND or LIMITED hmi level to application
+   * depends on application type
+   * @param app Application to deactivate
+   */
+  virtual void DeactivateApp(ApplicationSharedPtr app,
```
Comment (AKalinich-Luxoft) [inline src/components/include/application_manager/state_controller.h:232]: @AByzhynar missing description of window_id param

File: src/components/application_manager/src/state_controller_impl.cc [hunk 6]
```diff
@@ -1270,15 +1270,7 @@ mobile_apis::VideoStreamingState::eType StateControllerImpl::CalcVideoState(
     const mobile_apis::HMILevel::eType hmi_level) const {
   auto state = mobile_apis::VideoStreamingState::NOT_STREAMABLE;
 
-  if (app->webengine_projection_enabled()) {
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/src/state_controller_impl.cc:1273]: @AByzhynar why did you remove this? Wrong behavior found by a new unit tests?

Comment (AByzhynar) [reply src/components/application_manager/src/state_controller_impl.cc:1273]: @AKalinich-Luxoft It was redundant at it is covered by IsVideoApplication method

File: src/components/application_manager/test/application_manager_impl_test.cc (hunk 4 above)
Comment (AByzhynar) [reply src/components/application_manager/test/application_manager_impl_test.cc]: @AKalinich-Luxoft Done in e570bb7a3d5d48b9e64a5172d70804c532ec880f

File: src/components/application_manager/test/application_manager_impl_test.cc (hunk 2 above)
Comment (AByzhynar) [reply src/components/application_manager/test/application_manager_impl_test.cc]: @AKalinich-Luxoft Done in e570bb7a3d5d48b9e64a5172d70804c532ec880f
Thank you

File: src/components/application_manager/test/application_manager_impl_test.cc (hunk 1 above)
Comment (AByzhynar) [reply src/components/application_manager/test/application_manager_impl_test.cc]: @mvorobio Thank you. The functions separation was done in  e570bb7a3d5d48b9e64a5172d70804c532ec880f

File: src/components/application_manager/test/application_manager_impl_test.cc (hunk 3 above)
Comment (AByzhynar) [reply src/components/application_manager/test/application_manager_impl_test.cc]: @AKalinich-Luxoft Thank you. I found a better way) See 9a4d4bb83b71d728d39c231e38bca1bb680fb997

File: src/components/include/application_manager/state_controller.h (hunk 5 above)
Comment (AByzhynar) [reply src/components/include/application_manager/state_controller.h:233]: @AKalinich-Luxoft Thank you. Done in 6467b21e3df1cb7c9fc032b91f45d27e042b0063
